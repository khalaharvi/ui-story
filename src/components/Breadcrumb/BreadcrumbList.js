import styled from 'react-emotion';

const BreadcrumbList = styled('ul')({
  margin: 0,
  padding: 0,
  display: 'flex',

  listStyleType: 'none'
});

export default BreadcrumbList;
