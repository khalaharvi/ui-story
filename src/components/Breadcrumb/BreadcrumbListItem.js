import styled from 'react-emotion';
import _get from 'lodash.get';

const BreadcrumbListItem = styled('li')(({ theme }) => ({
  marginRight: '10px',
  color: _get(theme, 'text.secondary', 'black'),
  '&:last-child': {
    marginRight: 0
  }
}));

export default BreadcrumbListItem;
