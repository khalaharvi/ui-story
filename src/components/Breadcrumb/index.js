export { default, getLinkStyle } from './Breadcrumb';
