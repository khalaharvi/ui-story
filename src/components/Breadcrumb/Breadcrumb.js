import React from 'react';
import PropTypes from 'prop-types';
import { css, cx as emotionCx } from 'react-emotion';
import { withTheme } from 'emotion-theming';
import _get from 'lodash.get';

import { createClassNames } from 'utils';
import Link from 'components/Link';
import BreadcrumbList from './BreadcrumbList';
import BreadcrumbListItem from './BreadcrumbListItem';
import BreadcrumbSeparator from './BreadcrumbSeparator';

const cx = createClassNames('breadcrumb');

export const getLinkStyle = theme => css({
  color: _get(theme, 'text.secondary'),
  textDecoration: 'none',

  '&:hover': {
    color: _get(theme, 'text.default')
  }
});

// TO DO: add SEO rich markup
const Breadcrumb = ({
  items,
  separator,
  rootURL,
  rootLabel,
  rootComponent: RootComponent,
  itemComponent: ItemComponent,
  theme
}) => {
  const link = getLinkStyle(theme);

  const rootPath = (
    <BreadcrumbListItem className={cx('item')}>
      <RootComponent className={emotionCx(cx('link'), link)} label={rootLabel} href={rootURL} />
    </BreadcrumbListItem>
  );

  const breadcrumb = items.map((item, idx) => {
    const isLast = (idx === items.length - 1);
    const cleanSeparator = typeof separator === 'string' ? separator.trim() : separator;

    return (
      <BreadcrumbListItem className={cx('item', isLast && 'item--selected')} key={`${item.href}`}>
        <BreadcrumbSeparator className={cx('separator')}>{cleanSeparator}</BreadcrumbSeparator>
        {!isLast ? (
          <ItemComponent item={item} className={emotionCx(cx('link'), link)} />
        ) : (
          item.label
        )}
      </BreadcrumbListItem>
    );
  });

  return (
    <div className={cx('')}>
      <BreadcrumbList className={cx('list')}>
        {rootPath}
        {breadcrumb}
      </BreadcrumbList>
    </div>
  );
};

Breadcrumb.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      href: PropTypes.string.isRequired
    })
  ),
  separator: PropTypes.node,
  rootURL: PropTypes.string,
  rootLabel: PropTypes.string,
  rootComponent: PropTypes.func,
  itemComponent: PropTypes.func
};

Breadcrumb.defaultProps = {
  items: [],
  separator: ' > ',
  rootLabel: 'Home',
  rootURL: null,
  rootComponent: ({ href, label, ...props }) => (
    <Link to={href} {...props}>{label}</Link>
  ),
  itemComponent: ({ item, ...props }) => (
    <Link to={item.href} {...props}>{item.label}</Link>
  )
};

export default withTheme(Breadcrumb);
