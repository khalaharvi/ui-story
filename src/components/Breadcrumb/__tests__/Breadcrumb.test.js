import React from 'react';
import { mountWithTheme } from 'utils/testing';

import Breadcrumb from '../Breadcrumb';

describe('<Breadcrumb />', () => {
  it('renders correctly', () => {
    const wrapper = mountWithTheme(
      <Breadcrumb
        items={[
          {
            href: '/white',
            label: 'white'
          },
          {
            href: '/white/white1',
            label: 'white1'
          },
          {
            href: 'white/white1/white11',
            label: 'white1.1'
          }
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly without items', () => {
    const wrapper = mountWithTheme(<Breadcrumb />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with rootLabel', () => {
    const wrapper = mountWithTheme(
      <Breadcrumb
        rootLabel="Homepage"
        items={[
          {
            href: '/white',
            label: 'white'
          },
          {
            href: '/white/white1',
            label: 'white1'
          },
          {
            href: 'white/white1/white11',
            label: 'white1.1'
          }
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('has a rootURL prop', () => {
    const rootLink = 'deringhall.com';

    const wrapper = mountWithTheme(
      <Breadcrumb
        rootURL={rootLink}
        items={[
          {
            href: '/white',
            label: 'white'
          },
          {
            href: '/white/white1',
            label: 'white1'
          },
          {
            href: 'white/white1/white11',
            label: 'white1.1'
          }
        ]}
      />
    );

    const breadcrumb = wrapper.find('ul');

    expect(breadcrumb.children()).toHaveLength(4);

    expect(
      wrapper
        .find('a')
        .first()
        .prop('href')
    ).toEqual('deringhall.com');

    wrapper.unmount();
  });

  it('has a separator prop that can be a custom component', () => {
    const wrapper = mountWithTheme(
      <Breadcrumb
        separator={<span role="img" aria-label="separator">üîç</span>}
        items={[
          {
            href: '/white',
            label: 'white'
          },
          {
            href: '/white/white1',
            label: 'white1'
          },
          {
            href: 'white/white1/white11',
            label: 'white1.1'
          }
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('has a rootComponent prop that can be a custom component', () => {
    const wrapper = mountWithTheme(
      <Breadcrumb
        rootComponent={({ label }) => (<span>{label}</span>)}
        items={[
          {
            href: '/white',
            label: 'white'
          },
          {
            href: '/white/white1',
            label: 'white1'
          },
          {
            href: 'white/white1/white11',
            label: 'white1.1'
          }
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('has a itemComponent prop that can be a custom component', () => {
    const wrapper = mountWithTheme(
      <Breadcrumb
        itemComponent={({ item }) => (<span>{item.label}</span>)}
        items={[
          {
            href: '/white',
            label: 'white'
          },
          {
            href: '/white/white1',
            label: 'white1'
          },
          {
            href: 'white/white1/white11',
            label: 'white1.1'
          }
        ]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
