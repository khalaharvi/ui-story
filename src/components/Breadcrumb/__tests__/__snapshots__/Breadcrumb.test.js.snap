// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> has a itemComponent prop that can be a custom component 1`] = `
.emotion-2 {
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-21 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
}

.emotion-5 {
  margin-right: 10px;
  color: #6d6d6a;
}

.emotion-5:last-child {
  margin-right: 0;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-7 {
  margin-right: 10px;
  color: #6d6d6a;
}

<WithTheme(Breadcrumb)
  itemComponent={[Function]}
  items={
    Array [
      Object {
        "href": "/white",
        "label": "white",
      },
      Object {
        "href": "/white/white1",
        "label": "white1",
      },
      Object {
        "href": "white/white1/white11",
        "label": "white1.1",
      },
    ]
  }
>
  <Breadcrumb
    itemComponent={[Function]}
    items={
      Array [
        Object {
          "href": "/white",
          "label": "white",
        },
        Object {
          "href": "/white/white1",
          "label": "white1",
        },
        Object {
          "href": "white/white1/white11",
          "label": "white1.1",
        },
      ]
    }
    rootComponent={[Function]}
    rootLabel="Home"
    rootURL={null}
    separator=" > "
  >
    <div
      className="dh-breadcrumb"
    >
      <Styled(ul)
        className="dh-breadcrumb-list"
      >
        <ul
          className="dh-breadcrumb-list emotion-21 emotion-22"
        >
          <Styled(li)
            className="dh-breadcrumb-item"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <rootComponent
                className="dh-breadcrumb-link emotion-2"
                href={null}
                label="Home"
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to={null}
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href={null}
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href={null}
                      onClick={null}
                    >
                      Home
                    </a>
                  </Styled(a)>
                </Link>
              </rootComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  &gt;
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-2"
                item={
                  Object {
                    "href": "/white",
                    "label": "white",
                  }
                }
              >
                <span>
                  white
                </span>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white/white1"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  &gt;
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-2"
                item={
                  Object {
                    "href": "/white/white1",
                    "label": "white1",
                  }
                }
              >
                <span>
                  white1
                </span>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item dh-breadcrumb-item--selected"
            key="white/white1/white11"
          >
            <li
              className="dh-breadcrumb-item dh-breadcrumb-item--selected emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  &gt;
                </span>
              </Styled(span)>
              white1.1
            </li>
          </Styled(li)>
        </ul>
      </Styled(ul)>
    </div>
  </Breadcrumb>
</WithTheme(Breadcrumb)>
`;

exports[`<Breadcrumb /> has a rootComponent prop that can be a custom component 1`] = `
.emotion-0 {
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-25 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
}

.emotion-1 {
  margin-right: 10px;
  color: #6d6d6a;
}

.emotion-1:last-child {
  margin-right: 0;
}

.emotion-5 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 {
  margin-right: 10px;
  color: #6d6d6a;
}

<WithTheme(Breadcrumb)
  items={
    Array [
      Object {
        "href": "/white",
        "label": "white",
      },
      Object {
        "href": "/white/white1",
        "label": "white1",
      },
      Object {
        "href": "white/white1/white11",
        "label": "white1.1",
      },
    ]
  }
  rootComponent={[Function]}
>
  <Breadcrumb
    itemComponent={[Function]}
    items={
      Array [
        Object {
          "href": "/white",
          "label": "white",
        },
        Object {
          "href": "/white/white1",
          "label": "white1",
        },
        Object {
          "href": "white/white1/white11",
          "label": "white1.1",
        },
      ]
    }
    rootComponent={[Function]}
    rootLabel="Home"
    rootURL={null}
    separator=" > "
  >
    <div
      className="dh-breadcrumb"
    >
      <Styled(ul)
        className="dh-breadcrumb-list"
      >
        <ul
          className="dh-breadcrumb-list emotion-25 emotion-26"
        >
          <Styled(li)
            className="dh-breadcrumb-item"
          >
            <li
              className="dh-breadcrumb-item emotion-1 emotion-2"
            >
              <rootComponent
                className="dh-breadcrumb-link emotion-0"
                href={null}
                label="Home"
              >
                <span>
                  Home
                </span>
              </rootComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white"
          >
            <li
              className="dh-breadcrumb-item emotion-1 emotion-2"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-3 emotion-4"
                >
                  &gt;
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-0"
                item={
                  Object {
                    "href": "/white",
                    "label": "white",
                  }
                }
              >
                <Link
                  className="dh-breadcrumb-link emotion-0"
                  external={false}
                  noFollow={false}
                  to="/white"
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-0"
                    href="/white"
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-5 emotion-6"
                      href="/white"
                      onClick={null}
                    >
                      white
                    </a>
                  </Styled(a)>
                </Link>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white/white1"
          >
            <li
              className="dh-breadcrumb-item emotion-1 emotion-2"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-3 emotion-4"
                >
                  &gt;
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-0"
                item={
                  Object {
                    "href": "/white/white1",
                    "label": "white1",
                  }
                }
              >
                <Link
                  className="dh-breadcrumb-link emotion-0"
                  external={false}
                  noFollow={false}
                  to="/white/white1"
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-0"
                    href="/white/white1"
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-5 emotion-6"
                      href="/white/white1"
                      onClick={null}
                    >
                      white1
                    </a>
                  </Styled(a)>
                </Link>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item dh-breadcrumb-item--selected"
            key="white/white1/white11"
          >
            <li
              className="dh-breadcrumb-item dh-breadcrumb-item--selected emotion-1 emotion-2"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-3 emotion-4"
                >
                  &gt;
                </span>
              </Styled(span)>
              white1.1
            </li>
          </Styled(li)>
        </ul>
      </Styled(ul)>
    </div>
  </Breadcrumb>
</WithTheme(Breadcrumb)>
`;

exports[`<Breadcrumb /> has a separator prop that can be a custom component 1`] = `
.emotion-2 {
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-29 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
}

.emotion-5 {
  margin-right: 10px;
  color: #6d6d6a;
}

.emotion-5:last-child {
  margin-right: 0;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-7 {
  margin-right: 10px;
  color: #6d6d6a;
}

<WithTheme(Breadcrumb)
  items={
    Array [
      Object {
        "href": "/white",
        "label": "white",
      },
      Object {
        "href": "/white/white1",
        "label": "white1",
      },
      Object {
        "href": "white/white1/white11",
        "label": "white1.1",
      },
    ]
  }
  separator={
    <span
      aria-label="separator"
      role="img"
    >
      🔍
    </span>
  }
>
  <Breadcrumb
    itemComponent={[Function]}
    items={
      Array [
        Object {
          "href": "/white",
          "label": "white",
        },
        Object {
          "href": "/white/white1",
          "label": "white1",
        },
        Object {
          "href": "white/white1/white11",
          "label": "white1.1",
        },
      ]
    }
    rootComponent={[Function]}
    rootLabel="Home"
    rootURL={null}
    separator={
      <span
        aria-label="separator"
        role="img"
      >
        🔍
      </span>
    }
  >
    <div
      className="dh-breadcrumb"
    >
      <Styled(ul)
        className="dh-breadcrumb-list"
      >
        <ul
          className="dh-breadcrumb-list emotion-29 emotion-30"
        >
          <Styled(li)
            className="dh-breadcrumb-item"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <rootComponent
                className="dh-breadcrumb-link emotion-2"
                href={null}
                label="Home"
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to={null}
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href={null}
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href={null}
                      onClick={null}
                    >
                      Home
                    </a>
                  </Styled(a)>
                </Link>
              </rootComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  <span
                    aria-label="separator"
                    role="img"
                  >
                    🔍
                  </span>
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-2"
                item={
                  Object {
                    "href": "/white",
                    "label": "white",
                  }
                }
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to="/white"
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href="/white"
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href="/white"
                      onClick={null}
                    >
                      white
                    </a>
                  </Styled(a)>
                </Link>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white/white1"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  <span
                    aria-label="separator"
                    role="img"
                  >
                    🔍
                  </span>
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-2"
                item={
                  Object {
                    "href": "/white/white1",
                    "label": "white1",
                  }
                }
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to="/white/white1"
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href="/white/white1"
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href="/white/white1"
                      onClick={null}
                    >
                      white1
                    </a>
                  </Styled(a)>
                </Link>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item dh-breadcrumb-item--selected"
            key="white/white1/white11"
          >
            <li
              className="dh-breadcrumb-item dh-breadcrumb-item--selected emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  <span
                    aria-label="separator"
                    role="img"
                  >
                    🔍
                  </span>
                </span>
              </Styled(span)>
              white1.1
            </li>
          </Styled(li)>
        </ul>
      </Styled(ul)>
    </div>
  </Breadcrumb>
</WithTheme(Breadcrumb)>
`;

exports[`<Breadcrumb /> renders correctly 1`] = `
.emotion-2 {
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-29 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
}

.emotion-5 {
  margin-right: 10px;
  color: #6d6d6a;
}

.emotion-5:last-child {
  margin-right: 0;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-7 {
  margin-right: 10px;
  color: #6d6d6a;
}

<WithTheme(Breadcrumb)
  items={
    Array [
      Object {
        "href": "/white",
        "label": "white",
      },
      Object {
        "href": "/white/white1",
        "label": "white1",
      },
      Object {
        "href": "white/white1/white11",
        "label": "white1.1",
      },
    ]
  }
>
  <Breadcrumb
    itemComponent={[Function]}
    items={
      Array [
        Object {
          "href": "/white",
          "label": "white",
        },
        Object {
          "href": "/white/white1",
          "label": "white1",
        },
        Object {
          "href": "white/white1/white11",
          "label": "white1.1",
        },
      ]
    }
    rootComponent={[Function]}
    rootLabel="Home"
    rootURL={null}
    separator=" > "
  >
    <div
      className="dh-breadcrumb"
    >
      <Styled(ul)
        className="dh-breadcrumb-list"
      >
        <ul
          className="dh-breadcrumb-list emotion-29 emotion-30"
        >
          <Styled(li)
            className="dh-breadcrumb-item"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <rootComponent
                className="dh-breadcrumb-link emotion-2"
                href={null}
                label="Home"
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to={null}
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href={null}
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href={null}
                      onClick={null}
                    >
                      Home
                    </a>
                  </Styled(a)>
                </Link>
              </rootComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  &gt;
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-2"
                item={
                  Object {
                    "href": "/white",
                    "label": "white",
                  }
                }
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to="/white"
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href="/white"
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href="/white"
                      onClick={null}
                    >
                      white
                    </a>
                  </Styled(a)>
                </Link>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white/white1"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  &gt;
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-2"
                item={
                  Object {
                    "href": "/white/white1",
                    "label": "white1",
                  }
                }
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to="/white/white1"
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href="/white/white1"
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href="/white/white1"
                      onClick={null}
                    >
                      white1
                    </a>
                  </Styled(a)>
                </Link>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item dh-breadcrumb-item--selected"
            key="white/white1/white11"
          >
            <li
              className="dh-breadcrumb-item dh-breadcrumb-item--selected emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  &gt;
                </span>
              </Styled(span)>
              white1.1
            </li>
          </Styled(li)>
        </ul>
      </Styled(ul)>
    </div>
  </Breadcrumb>
</WithTheme(Breadcrumb)>
`;

exports[`<Breadcrumb /> renders correctly with rootLabel 1`] = `
.emotion-2 {
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-29 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
}

.emotion-5 {
  margin-right: 10px;
  color: #6d6d6a;
}

.emotion-5:last-child {
  margin-right: 0;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-7 {
  margin-right: 10px;
  color: #6d6d6a;
}

<WithTheme(Breadcrumb)
  items={
    Array [
      Object {
        "href": "/white",
        "label": "white",
      },
      Object {
        "href": "/white/white1",
        "label": "white1",
      },
      Object {
        "href": "white/white1/white11",
        "label": "white1.1",
      },
    ]
  }
  rootLabel="Homepage"
>
  <Breadcrumb
    itemComponent={[Function]}
    items={
      Array [
        Object {
          "href": "/white",
          "label": "white",
        },
        Object {
          "href": "/white/white1",
          "label": "white1",
        },
        Object {
          "href": "white/white1/white11",
          "label": "white1.1",
        },
      ]
    }
    rootComponent={[Function]}
    rootLabel="Homepage"
    rootURL={null}
    separator=" > "
  >
    <div
      className="dh-breadcrumb"
    >
      <Styled(ul)
        className="dh-breadcrumb-list"
      >
        <ul
          className="dh-breadcrumb-list emotion-29 emotion-30"
        >
          <Styled(li)
            className="dh-breadcrumb-item"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <rootComponent
                className="dh-breadcrumb-link emotion-2"
                href={null}
                label="Homepage"
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to={null}
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href={null}
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href={null}
                      onClick={null}
                    >
                      Homepage
                    </a>
                  </Styled(a)>
                </Link>
              </rootComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  &gt;
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-2"
                item={
                  Object {
                    "href": "/white",
                    "label": "white",
                  }
                }
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to="/white"
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href="/white"
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href="/white"
                      onClick={null}
                    >
                      white
                    </a>
                  </Styled(a)>
                </Link>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item"
            key="/white/white1"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  &gt;
                </span>
              </Styled(span)>
              <itemComponent
                className="dh-breadcrumb-link emotion-2"
                item={
                  Object {
                    "href": "/white/white1",
                    "label": "white1",
                  }
                }
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to="/white/white1"
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href="/white/white1"
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href="/white/white1"
                      onClick={null}
                    >
                      white1
                    </a>
                  </Styled(a)>
                </Link>
              </itemComponent>
            </li>
          </Styled(li)>
          <Styled(li)
            className="dh-breadcrumb-item dh-breadcrumb-item--selected"
            key="white/white1/white11"
          >
            <li
              className="dh-breadcrumb-item dh-breadcrumb-item--selected emotion-5 emotion-6"
            >
              <Styled(span)
                className="dh-breadcrumb-separator"
              >
                <span
                  className="dh-breadcrumb-separator emotion-7 emotion-8"
                >
                  &gt;
                </span>
              </Styled(span)>
              white1.1
            </li>
          </Styled(li)>
        </ul>
      </Styled(ul)>
    </div>
  </Breadcrumb>
</WithTheme(Breadcrumb)>
`;

exports[`<Breadcrumb /> renders correctly without items 1`] = `
.emotion-2 {
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-7 {
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
}

.emotion-5 {
  margin-right: 10px;
  color: #6d6d6a;
}

.emotion-5:last-child {
  margin-right: 0;
}

.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #6d6d6a;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<WithTheme(Breadcrumb)>
  <Breadcrumb
    itemComponent={[Function]}
    items={Array []}
    rootComponent={[Function]}
    rootLabel="Home"
    rootURL={null}
    separator=" > "
  >
    <div
      className="dh-breadcrumb"
    >
      <Styled(ul)
        className="dh-breadcrumb-list"
      >
        <ul
          className="dh-breadcrumb-list emotion-7 emotion-8"
        >
          <Styled(li)
            className="dh-breadcrumb-item"
          >
            <li
              className="dh-breadcrumb-item emotion-5 emotion-6"
            >
              <rootComponent
                className="dh-breadcrumb-link emotion-2"
                href={null}
                label="Home"
              >
                <Link
                  className="dh-breadcrumb-link emotion-2"
                  external={false}
                  noFollow={false}
                  to={null}
                >
                  <Styled(a)
                    className="dh-link dh-breadcrumb-link emotion-2"
                    href={null}
                    onClick={null}
                  >
                    <a
                      className="dh-link dh-breadcrumb-link emotion-0 emotion-1"
                      href={null}
                      onClick={null}
                    >
                      Home
                    </a>
                  </Styled(a)>
                </Link>
              </rootComponent>
            </li>
          </Styled(li)>
        </ul>
      </Styled(ul)>
    </div>
  </Breadcrumb>
</WithTheme(Breadcrumb)>
`;
